addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/handler/BridgeHandler.java

```diff
@@ -0,0 +1,377 @@
+        logger.info("Tankstelle {} was unregistered from config {} ", tankstelle.getUID().toString(),
+                getThing().getUID().toString());
+        tankstellenThingList.remove(tankstelle);
+    }
+
+    /***
+     * Updates the data from tankerkoenig api (no update on things)
+     */
+    public void updateTankstellenData() {
+        // Get data
+        try {
+            String locationIDsString = "";
+            if (useOpeningTime) {
+                logger.debug("Opening times are used");
+                locationIDsString = generateOpenLocationIDsString();
+            } else {
+                logger.debug("No opening times are used");
+                locationIDsString = generateLocationIDsString();
+            }
+            if (locationIDsString.isEmpty()) {
+                logger.info("No tankstellen id's found. Nothing to update");
+                return;
+            }
+            TankerkoenigService service = new TankerkoenigService();
+            TankerkoenigListResult result = service.getTankstellenListData(this.getApiKey(), locationIDsString);
+            if (!result.isOk()) {
+                // if the result is not OK, no updates are done and the status of the Bridge goes to unknown!
+                updateStatus(ThingStatus.UNKNOWN);
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+                setTankerkoenigListResult(result);
+                tankstellenList.clear();
+                for (LittleStation station : result.getPrices().getStations()) {
+                    tankstellenList.put(station.getID(), station);
+                }
+                logger.debug("UpdateTankstellenData: tankstellenList.size {}", tankstellenList.size());
+            }
+        } catch (ParseException e) {
+            logger.error("ParseException: {}", e.toString());
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    /***
+     * Updates all registered Tankstellen with new data
+     */
+    public void updateTankstellenThings() {
+        logger.debug("UpdateTankstellenThings: tankstellenThingList.size {}", tankstellenThingList.size());
+        for (Thing thing : tankstellenThingList) {
+            TankerkoenigHandler tkh = (TankerkoenigHandler) thing.getHandler();
+            LittleStation s = this.tankstellenList.get(tkh.getLocationID());
+            if (s == null) {
+                logger.debug("Tankstelle with id {}  is not updated, because it is not open!", tkh.getLocationID());
+            } else {
+                tkh.updateData(s);
+            }
+        }
+    }
+
+    /***
+     * Generates a comma separated string with all tankstellen id's
+     *
+     * @return
+     */
+    private String generateLocationIDsString() {
+        StringBuilder sb = new StringBuilder();
+        for (Thing thing : tankstellenThingList) {
+            TankerkoenigHandler tkh = (TankerkoenigHandler) thing.getHandler();
+            if (sb.toString().isEmpty()) {
```

Comment:
Will do, you pointed me to the .isempty method on another location and I used it here as well. Will change back.